avahi-common/alternative.c

```diff
@@ -49,15 +49,20 @@ static void drop_incomplete_utf8(char *c) {
 }
 
 char *avahi_alternative_host_name(const char *s) {
+    char label[AVAHI_LABEL_MAX], alternative[AVAHI_LABEL_MAX*4+1];
```

Comment:
I think it should be `(AVAHI_LABEL_MAX-1)*4+1` but it doesn't matter much probably.

Reply:
I guess it can be changed in a separate commit where something like `AVAHI_LABEL_ESCAPED_MAX` can be introduced. It can be used to extract hostnames properly later.